Kuadrant/multi-cluster-traffic-controller
# Development experience improvements

## Description

Changes to _slightly_ improve the development experience

### 1. Webhook base template

The webhook configuration is generated by setting the CA bundle into an existing template and applying it into the workload clusters. Currently the file is checked in version control which means the changed file might accidentally be commited. In order to avoid this, keep a separate base template checked in, and the effectively used one in .gitignore

### 2. Automatically generate the control plane secret for the agent

Currently the secret needs to be created manually. There is some existing functionality to create these type of secrets so refactor that functionality to be able to use it when deploying the agent into the cluster

## Verification steps

In order to verify these changes, follow the steps in the [README](https://github.com/Kuadrant/multi-cluster-traffic-controller/blob/9f43933c44e798cfa12458f0e936b8203a69874b/README.md) to deploy the agent and verify that:
1. Git doesn't detect any changes related to the webhook configuration
2. The secret for the agent has been created successfully
    ```sh
    kubectl get secret/control-plane-cluster -n mctc-system -o yaml
    ```
